customFunc.py

```
@@ -251,7 +221,31 @@ def scroll_into_view(driver, element):
     print("scrolling into view element with id " + element.id)
     driver.execute_script("arguments[0].scrollIntoView(true);", element)
 
-    
+
+class gmail_checker
+    def __init__(self):
+        self.service = auth.get_service_gmail()  # open gmail API client
+
+    def count(self,email,needle)
+        """
+        Search in user_id's email for string
+        Accept:
+        email - user's email to search in it
+        needle - string to search
+
+        Returns:
+        number of email messages found
+        """
+        try:
+            search_id = service.users().messages().list(userId=user_id, q=search_string).execute()
+            number_results = int(search_id['resultSizeEstimate'])
+
+            return number_results
+
+        except (errors.HttpError, errors):
+            return 'error: %s' % errors

```

please always `println` when you catch an exception -
it's vital to understand in Jenkins what went wrong.

(BTW - I want to add a proper logging library sometime, and when you add such - you would `log.error()` in case of exception)

---

Also, I think you should return `0` in this case.
Always return the same type.

in this case, if you return a string - 
What would the code receiving it would do with it?
I have no idea how would this behave:
`while max_wait > 0 and num_emails_received < 1: `

---

you're right. i want to throw the exception and catch it outside